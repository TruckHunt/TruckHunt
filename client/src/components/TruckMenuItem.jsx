import React from 'react';
import { Col, Panel, Row } from 'react-bootstrap';

const TruckMenuItem = ({ item, onAddToCartClicked }) => {
  const { name, item_description, price } = item;
  return (
    <Col md={6}>
      <Panel>
        <Row>
          <Col md={9} sm={9} xs={9}>
            <h5> {name} </h5>
            <p> {item_description} </p>
          </Col>
          <Col md={3} sm={9} xs={3}>
            <p> &#36;{price} </p>
            <button onClick={onAddToCartClicked}>
              Add to cart
            </button>
          </Col>
        </Row>
      </Panel>
    </Col>
  );
};

export default TruckMenuItem;
