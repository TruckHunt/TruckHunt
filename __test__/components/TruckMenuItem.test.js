import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom';
import TruckMenuItem from '../../client/src/components/TruckMenuItem';

const menuItem = {
  menu_item_id: 36,
  vendor_id: null,
  name: 'All American Club',
  course: 'Wraps',
  food_category: 'Cold Truck: Pre-packaged sandwiches: snacks: fruit: various beverages',
  price: 775,
  item_description: 'Roast turkey, honey ham, bacon, Monterey Jack cheese, lettuce, tomato and dijonaisse'
};

describe('Truck Menu Item  Component Tests', () => {
  test('Truck Menu Item matches snapshot', () => {
    const component = renderer.create(
      <StaticRouter>
        <TruckMenuItem
          item={menuItem}
        />
      </StaticRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('TruckMenuItem Component Test', () => {
  let truckMenuItem;
  const item = {
    name: 'salad',
    price: 499
  };

  beforeEach(() => {
    truckMenuItem = mount(
      <StaticRouter>
        <TruckMenuItem item={item} />
      </StaticRouter>
    );
  });

  it('should create a TruckMenuItem component', () => {
    expect(truckMenuItem.exists()).toBe(true);
  });

  it('should contain the name, description and price', () => {
    const text = truckMenuItem.text();
    expect(text).toContain(item.name);
  });
});
