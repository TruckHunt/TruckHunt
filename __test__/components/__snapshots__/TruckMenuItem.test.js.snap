// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Truck Menu Item  Component Tests Truck Menu Item matches snapshot 1`] = `
<div
  className="ui orange card animated slideInUp"
  href={undefined}
  onClick={[Function]}
>
  <div
    className="content"
  >
    <div
      className="header"
    >
      All American Club
    </div>
    <div
      className="meta"
    >
      $
      7.75
    </div>
    <div
      className="description"
    >
      Roast turkey, honey ham, bacon, Monterey Jack cheese, lettuce, tomato and dijonaisse
    </div>
  </div>
  <div
    className="extra content"
  >
    <a
      href="/truckMenu"
      onClick={[Function]}
    >
      <button
        className="ui orange mini basic icon button right floated"
        onClick={[Function]}
        tabIndex={undefined}
      >
        <i
          aria-hidden="true"
          className="plus icon"
        />
         
      </button>
    </a>
  </div>
</div>
`;
